src/pkg/scan/vuln/summary.go

```diff
@@ -17,21 +17,76 @@ package vuln
 import (
 	"time"
 
+	"github.com/goharbor/harbor/src/jobservice/job"
 	v1 "github.com/goharbor/harbor/src/pkg/scan/rest/v1"
 )
 
 // NativeReportSummary is the default supported scan report summary model.
 // Generated based on the report with v1.MimeTypeNativeReport mime type.
 type NativeReportSummary struct {
-	ReportID    string                `json:"report_id"`
-	ScanStatus  string                `json:"scan_status"`
-	Severity    Severity              `json:"severity"`
-	Duration    int64                 `json:"duration"`
-	Summary     *VulnerabilitySummary `json:"summary"`
-	CVEBypassed []string              `json:"-"`
-	StartTime   time.Time             `json:"start_time"`
-	EndTime     time.Time             `json:"end_time"`
-	Scanner     *v1.Scanner           `json:"scanner,omitempty"`
+	ReportID        string                `json:"report_id"`
+	ScanStatus      string                `json:"scan_status"`
+	Severity        Severity              `json:"severity"`
+	Duration        int64                 `json:"duration"`
+	Summary         *VulnerabilitySummary `json:"summary"`
+	CVEBypassed     []string              `json:"-"`
+	StartTime       time.Time             `json:"start_time"`
+	EndTime         time.Time             `json:"end_time"`
+	Scanner         *v1.Scanner           `json:"scanner,omitempty"`
+	CompletePercent int                   `json:"complete_percent"`
+
+	TotalCount    int `json:"-"`
+	CompleteCount int `json:"-"`
+}
+
+// Merge ...
+func (sum *NativeReportSummary) Merge(another *NativeReportSummary) *NativeReportSummary {
+	r := &NativeReportSummary{}
+
+	r.ReportID = sum.ReportID
+	r.StartTime = minTime(sum.StartTime, another.StartTime)
+	r.EndTime = maxTime(sum.EndTime, another.EndTime)
+	r.Duration = r.EndTime.Unix() - r.StartTime.Unix()
+	r.Scanner = sum.Scanner
+	r.TotalCount = sum.TotalCount + another.TotalCount
+	r.CompleteCount = sum.CompleteCount + another.CompleteCount
+	r.CompletePercent = r.CompleteCount * 100 / r.TotalCount
+
+	if sum.Severity.String() != "" && another.Severity.String() != "" {
```

Comment:
add a compare method in the model of `Severity`?

Reply:
There is already exist a method `Code` to compare the severity, but not work in this scenario.

When the severity is empty string, it's bigger than all others severity. So in this scenario we should first ensure that the severities are not empty string then compare thire values and chose the bigger one.